zebra/zebra_rib.c

```diff
@@ -2224,6 +2226,8 @@ int rib_add_multipath(afi_t afi, safi_t safi, struct prefix *p,
 		if (re->type == ZEBRA_ROUTE_BGP
 		    && CHECK_FLAG(re->flags, ZEBRA_FLAG_IBGP))
 			re->distance = 200;
+		else if (re->type == ZEBRA_ROUTE_KERNEL)
+			re->distance = re->metric;
```

Comment:
Maybe, but right know I fail to see how can I insert a route into the kernel from a random daemon and not have zebra take it as the most important route in the system. I am thinking backup routes for routes learned some other way.